processor/src/main/java/org/mapstruct/ap/internal/model/LifecycleMethodResolver.java

```diff
@@ -144,6 +144,16 @@ private static List<LifecycleCallbackMethodReference> collectLifecycleCallbackMe
             method.getReturnType(),
             SelectionCriteria.forLifecycleMethods( selectionParameters ) );
 
+        if ( matchingMethods.isEmpty() ) {
```

Comment:
Instead of redoing this with the `getResultType`. Can't we do it with the `getResultType`  from the start. I guess that's always populated, right?

```java
    @Override
    public Type getResultType() {
        return mappingTargetParameter != null ? mappingTargetParameter.getType() : returnType;
    }

```

Reply:
Will attempt this, this should also fix that what I did at the other remark is not necessary.